CodeEdit/Features/NavigatorArea/ProjectNavigator/OutlineView/ProjectNavigatorViewController+NSOutlineViewDataSource.swift

```diff
@@ -0,0 +1,119 @@
+//
+//  ProjectNavigatorViewController+NSOutlineViewDataSource.swift
+//  CodeEdit
+//
+//  Created by Khan Winter on 7/13/24.
+//
+
+import AppKit
+
+extension ProjectNavigatorViewController: NSOutlineViewDataSource {
+    func outlineView(_ outlineView: NSOutlineView, numberOfChildrenOfItem item: Any?) -> Int {
+        if let item = item as? CEWorkspaceFile {
+            return item.isFolder ? workspace?.workspaceFileManager?.childrenOfFile(item)?.count ?? 0 : 0
+        }
+        return content.count
+    }
+
+    func outlineView(_ outlineView: NSOutlineView, child index: Int, ofItem item: Any?) -> Any {
+        if let item = item as? CEWorkspaceFile,
+           let children = workspace?.workspaceFileManager?.childrenOfFile(item) {
+            return children[index]
+        }
+        return content[index]
+    }
+
+    func outlineView(_ outlineView: NSOutlineView, isItemExpandable item: Any) -> Bool {
+        if let item = item as? CEWorkspaceFile {
+            return item.isFolder
+        }
+        return false
+    }
+
+    /// write dragged file(s) to pasteboard
```

Comment:
Can we capitalize the first character for these 3 functions below

Reply:
iirc we create function names like `outlineView` classes should start with an uppercase character e.g. `ProjectNavigatorViewController`.
Or am I wrong here?

Reply:
@0xWDG you're not wrong I think he's talking about the commends. @FastestMolasses these weren't modified, just moved so I'd rather not touch them in this PR.

Reply:
Yep my bad I meant the function comments